gallery/visualtests/nxTextInput.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+const { Target } = require('@applitools/eyes-webdriverio');
+const { focusTest, hoverTest, simpleTest } = require('./testUtils');
+
+describe('NxTextInput', function() {
+  const simpleComponentSelector = '#nx-text-input-simple-example .nx-text-input',
+      validatableComponentSelector = '#nx-text-input-validation-example .nx-text-input',
+      passwordComponentSelector = '#nx-text-input-password-example .nx-text-input',
+      textareaComponentSelector = '#nx-text-input-textarea-validation-example .nx-text-input',
+      validatableTextareaComponentSelector = '#nx-text-input-textarea-validation-example .nx-text-input',
+      disabledComponentSelector = '#nx-text-input-disabled-example .nx-text-input.pristine',
+      disabledValidComponentSelector = '#nx-text-input-disabled-example .nx-text-input.valid',
+      disabledInvalidComponentSelector = '#nx-text-input-disabled-example .nx-text-input.invalid';
+
+  function getInputElementSelector(componentSelector, inputType = 'input') {
+    return `${componentSelector} ${inputType}`;
+  }
+
+  beforeEach(async function() {
+    await browser.url('#/pages/NxTextInput');
+  });
+
+  describe('Simple NxTextInput', function() {
+    it('has a light border when pristine', simpleTest(simpleComponentSelector));
+
+    it('has a dark border when hovered',
+        hoverTest(simpleComponentSelector, getInputElementSelector(simpleComponentSelector)));
```

Comment:
Basically the same concern as above: how is `hoverTest(selector, hoverSelector)` testing for a "dark border when hovered"?

Reply:
> Basically the same concern as above: how is hoverTest(selector, hoverSelector) testing for a "dark border when hovered"?

> It's up to the writer of the test to then go into applitools and verify that the initial screenshot of a newly written test matches the condition described in the test name.